### PR TITLE
AMI Recipe (Short UTT checker)

### PR DESCRIPTION
**Pull request by ftshijt**
Add a checker to detect if the speech signal is too short for subsampling. Those utts will result in an empty hypothesis.

This PR is proposed as a solution to #2746

### CODE REVIEW

**Comment by ftshijt** [comment]
The followings are the results I got using the default setting.


## asr_train_asr_raw_char_sp
### WER

|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
|---|---|---|---|---|---|---|---|---|
|decode_asr_lm_lm_train_lm_char_valid.loss.ave_asr_model_valid.acc.ave/ihm_dev|13057|94802|65.8|26.0|8.2|6.0|40.2|71.3|
|decode_asr_lm_lm_train_lm_char_valid.loss.ave_asr_model_valid.acc.ave/ihm_eval|12643|89666|63.8|28.6|7.5|5.9|42.0|68.5|

### CER

|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
|---|---|---|---|---|---|---|---|---|
|decode_asr_lm_lm_train_lm_char_valid.loss.ave_asr_model_valid.acc.ave/ihm_dev|13057|451641|80.7|8.8|10.5|5.7|25.1|71.3|
|decode_asr_lm_lm_train_lm_char_valid.loss.ave_asr_model_valid.acc.ave/ihm_eval|12643|432094|79.8|10.0|10.1|5.7|25.8|68.5|

File: espnet2/bin/asr_inference.py [hunk 1]
```diff
@@ -309,7 +310,13 @@ def inference(
             batch = {k: v[0] for k, v in batch.items() if not k.endswith("_lengths")}
 
             # N-best list of (text, token, token_int, hyp_object)
-            results = speech2text(**batch)
+            try:
+                results = speech2text(**batch)
+            except TooShortUttError:
+                logging.info(
```
Comment (sw005320) [inline espnet2/bin/asr_inference.py:316]:
```suggestion
                logging.warning(
```

File: espnet2/bin/asr_inference.py [hunk 2]
```diff
@@ -309,7 +310,13 @@ def inference(
...
+                    "Utterance is too short for subsampling, return empty results"
```
Comment (sw005320) [inline espnet2/bin/asr_inference.py:317]: Is it possible to add the following information?
1. utterance ID (if possible)
2. how many (subsampled?) frames of the utterance ID
3. how many (subsampled?) frames are required?

e.g, "Utterance [utterance ID] has XXX frames and is too short for subsampling (it needs more than YYY frames), return empty results"

**Review by sw005320 — COMMENTED** [review]
Many thanks!
- Can you also put "try except" in the espnet1 case as well? (I think it may happen in espnet1, right?)

@kamo-naoyuki, could you also check it?
I'm actually not sure where we should put this "try except".

File: espnet2/bin/asr_inference.py (hunk 2 above)
Comment (ftshijt) [reply espnet2/bin/asr_inference.py:317]: Sure, let me have a try on it

**Comment by ftshijt** [comment]
I have written a code for espnet1 but seems to need some time for testing them. Maybe we could merge it first so that you could go on to the next stage.

**Review by kamo-naoyuki — APPROVED** [review]
I think this is not the best way, but acceptable.

We need same workaround for

- VGG-RNN encoder
- STFT in frontend

and etc. This is TODO, i think it's okay for now.  I'm not sure we really should always do it when we'll add new module.

**Pull request merged**